From: Boyuan Yang <byang@debian.org>
Date: Mon, 13 May 2024 23:40:32 -0400
Subject: CMakeLists.txt: Do not enable NEON for armel armhf

According to https://wiki.debian.org/ArchitectureSpecificsMemo#armel
the armhf architecture does not guarantee NEON. Do not enable NEON
on armhf. Also disable NEON for armel to prevent FTBFS.
---
 CMakeLists.txt | 18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6aed6a9..f51166b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -85,16 +85,16 @@ SET(ly_lib_parts $<TARGET_OBJECTS:${ly_lib_name}_common_objects>)
 if(NOT MSVC)
   STRING(TOLOWER "${CMAKE_SYSTEM_PROCESSOR}" arch_lowercase)
 
+# Debian-specific: Do not enable Arm Neon kernels on non-arm64 architectures.
   if(arch_lowercase MATCHES "^arm" AND NOT arch_lowercase STREQUAL "arm64")
-    # Enable Arm Neon kernels.
-    ADD_DEFINITIONS(-DLIBYUV_NEON=1)
-    ADD_LIBRARY(${ly_lib_name}_neon OBJECT
-      ${ly_src_dir}/compare_neon.cc
-      ${ly_src_dir}/rotate_neon.cc
-      ${ly_src_dir}/row_neon.cc
-      ${ly_src_dir}/scale_neon.cc)
-    TARGET_COMPILE_OPTIONS(${ly_lib_name}_neon PRIVATE -mfpu=neon)
-    LIST(APPEND ly_lib_parts $<TARGET_OBJECTS:${ly_lib_name}_neon>)
+    MESSAGE("Debian-specific: Not enabling NEON on ${arch_lowercase}.")
+#    ADD_LIBRARY(${ly_lib_name}_neon OBJECT
+#      ${ly_src_dir}/compare_neon.cc
+#      ${ly_src_dir}/rotate_neon.cc
+#      ${ly_src_dir}/row_neon.cc
+#      ${ly_src_dir}/scale_neon.cc)
+#    TARGET_COMPILE_OPTIONS(${ly_lib_name}_neon PRIVATE -mfpu=neon)
+#    LIST(APPEND ly_lib_parts $<TARGET_OBJECTS:${ly_lib_name}_neon>)
   endif()
 
   if(arch_lowercase STREQUAL "aarch64" OR arch_lowercase STREQUAL "arm64")
